src/impute_srp_model.cpp

```diff
@@ -148,9 +148,29 @@ DataFrame impute_srp_model(
           }
         }
         if (from(i) == "response") {
-
-          // todo
-
+          if (from[i - 1] != "stable" ||
+              subject_id[i - 1] != subject_id[i]) {
+            stop("Unexpected");
+          }
+          double t_response = rtruncweibull(
+            shape[iter][g][0], scale[iter][g][0], t_min(i-1), t_max(i-1)
+          );
+          double dt_progression = rtruncweibull(
+              shape[iter][g][2], scale[iter][g][2],
+                                               t_min(i) - t_response, R_PosInf
+          );
+          double n_visits_prog = floor(dt_progression / visit_spacing(g));
```

Comment:
good point +1. Also, we might need to pre-allocate the _out vectors to avoid dynamic allocation of memory. Let's do that in a next PR though so that we don't pack too much in this one.